raiden/transfer/mediated_transfer/target.py

```diff
@@ -145,8 +145,9 @@ def handle_secretreveal(
 ):
     """ Validates and handles a ReceiveSecretReveal state change. """
     valid_secret = state_change.secrethash == target_state.transfer.lock.secrethash
+    should_send_secret = target_state.state == 'secret_request'
 
-    if valid_secret:
+    if valid_secret and should_send_secret:
```

Comment:
Why is the condition necessary here? And from my understanding `waiting_for_secret` would be a better name for the variable!?

Reply:
> Why is the condition necessary here? 

This is guarding the event `SendRevealSecret`, so that we only register the secret and send the secret reveal message once.

> And from my understanding waiting_for_secret would be a better name for the variable!?

This PR fixes the bug where a node spams another with secret reveal messages, because of that the variable was named `should_send_secret`, because we only want to do it once. It just happens that for the existing states we only send the reveal when the target is at `secret_request` state.